In [1]:
import pandas as pd
import json

df = pd.read_csv("C:/Users/kevin/ETL/Postgres/kaggle/glassdoor.csv", delimiter = ',')

In [2]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 165290 entries, 0 to 165289
Columns: 163 entries, benefits.benefitRatingDecimal to wwfu
dtypes: bool(11), float64(35), int64(23), object(94)
memory usage: 193.4+ MB


Este dataset dispone de 165.290 filas con 163 columnas, de lso cuales hay 23 columnas de numeros enteros, 35 decimales, 11 booleanos y 94 tipo objeto (texto).

In [10]:
df.head()

benefits.benefitRatingDecimal  benefits.comments  benefits.highlights  \
0                            0.0             2801.0               2801.0   
1                            4.0             2802.0               2802.0   
2                            0.0             2803.0               2803.0   
3                            0.0             2804.0               2804.0   
4                            0.0             2805.0               2805.0   

   benefits.numRatings benefits.employerSummary  breadCrumbs  \
0                    0                      NaN         2801   
1                    3                      NaN         2802   
2                    0                      NaN         2803   
3                    0                      NaN         2804   
4                    0                      NaN         2805   

   gaTrackerData.category  gaTrackerData.empId  gaTrackerData.empName  \
0                      -1              2183810            Modus Group   
1                      -1                10416   Boehringer Ingelheim   
2                      -1              1043373        Immobiliare.it    
3                      -1              2226886                  Codup   
4                      -1               851555  Markelytics Solutions   

  gaTrackerData.empSize  gaTrackerData.expired  \
0              501-1000                  False   
1              10000--1                  False   
2               201-500                  False   
3                  1-50                  False   
4               201-500                  False   

                  gaTrackerData.industry  gaTrackerData.industryId  \
0  Investment Banking & Asset Management                    200053   
1              Biotech & Pharmaceuticals                    200021   
2                                    NaN                         0   
3                                    NaN                         0   
4                             Consulting                    200028   

   gaTrackerData.jobId.long  gaTrackerData.jobId.int  \
0              3.227100e+09                      NaN   
1              3.406582e+09                      NaN   
2              3.230738e+09                      NaN   
3              3.406677e+09                      NaN   
4              3.201516e+09                      NaN   

                              gaTrackerData.jobTitle gaTrackerData.location  \
0                 Biogas Project Development Manager                 Warsaw   
1  Quality Manager - Boehringer Ingelheim Healthc...                    NaN   
2     Senior Software Engineer (PHP, Elixir, Python)                 Milano   
3                                Senior SQA Engineer                Karachi   
4                                   Research Manager              Bengaluru   

   gaTrackerData.locationId gaTrackerData.locationType  \
0                   3094484                       CITY   
1                   3183562                       CITY   
2                   2802090                       CITY   
3                   3212296                       CITY   
4                   2940587                       CITY   

  gaTrackerData.pageRequestGuid.guid  gaTrackerData.pageRequestGuid.guidValid  \
0   0000016e56d0b5f2b16c3aea720c55ae                                     True   
1   0000016e56d0b5cfa19a4b2ba225a2ee                                     True   
2   0000016e56d0b5c58f39733898e10986                                     True   
3   0000016e56d0b5bca1733da6de4f4098                                     True   
4   0000016e56d0b5ba9a6c4efc3844657c                                     True   

   gaTrackerData.pageRequestGuid.part1  gaTrackerData.pageRequestGuid.part2  \
0                        1573414548978                 -5662085852901386834   
1                        1573414548943                 -6802041636413660434   
2                        1573414548933                 -8126337365693101690   
3                        15734145489

In [5]:
df.dtypes

benefits.benefitRatingDecimal                                   float64
benefits.comments                                               float64
benefits.highlights                                             float64
benefits.numRatings                                               int64
benefits.employerSummary                                         object
breadCrumbs                                                       int64
gaTrackerData.category                                            int64
gaTrackerData.empId                                               int64
gaTrackerData.empName                                            object
gaTrackerData.empSize                                            object
gaTrackerData.expired                                              bool
gaTrackerData.industry                                           object
gaTrackerData.industryId                                          int64
gaTrackerData.jobId.long                                        

In [9]:
df.describe()

benefits.benefitRatingDecimal  benefits.comments  benefits.highlights  \
count                  165290.000000      165289.000000        165289.000000   
mean                        1.387690       82645.000000         82645.000000   
std                         1.826381       47714.968659         47714.968659   
min                         0.000000           1.000000             1.000000   
25%                         0.000000       41323.000000         41323.000000   
50%                         0.000000       82645.000000         82645.000000   
75%                         3.504717      123967.000000        123967.000000   
max                         5.000000      165289.000000        165289.000000   

       benefits.numRatings    breadCrumbs  gaTrackerData.category  \
count        165290.000000  165290.000000           165290.000000   
mean             17.650342   82645.500000             2476.188614   
std              49.901598   47715.257335             6098.395811   
min               0.000000       1.000000               -1.000000   
25%               0.000000   41323.250000               -1.000000   
50%               0.000000   82645.500000               -1.000000   
75%               6.000000  123967.750000               -1.000000   
max             596.000000  165290.000000            30165.000000   

       gaTrackerData.empId  gaTrackerData.industryId  \
count         1.652900e+05             165290.000000   
mean          6.945812e+05             142681.566731   
std           8.862075e+05              90479.876651   
min           0.000000e+00                  0.000000   
25%           5.632000e+03                  0.000000   
50%           3.043920e+05             200048.000000   
75%           1.107358e+06             200063.000000   
max           3.102014e+06             200147.000000   

       gaTrackerData.jobId.long  gaTrackerData.jobId.int  \
count              1.650440e+05             2.460000e+02   
mean               3.329226e+09             1.754135e+09   
std                1.167657e+08             3.473246e+08   
min                2.208549e+09             3.276949e+08   
25%                3.303018e+09             1.756631e+09   
50%                3.370946e+09             1.904030e+09   
75%                3.399873e+09             1.957060e+09   
max                3.412782e+09             2.022056e+09   

       gaTrackerData.locationId  gaTrackerData.pageRequestGuid.part1  \
count              1.652900e+05                         1.652900e+05   
mean               2.548963e+06                         1.573422e+12   
std                1.116096e+06                         9.163820e+06   
min               -1.000000e+00                         1.573401e+12   
25%                2.302160e+06                         1.573415e+12   
50%                2.739035e+06                         1.573425e+12   
75%                3.174503e+06                         1.573429e+12   
max                5.021990e+06                         1.573433e+12   

       gaTrackerData.pageRequestGuid.part2  gaTrackerData.sectorId  \
count                         1.652900e+05           165290.000000   
mean                         -6.921523e+18             7142.386557   
std                           1.327411e+18             4527.120950   
min                          -9.223346e+18                0.000000   
25%                          -8.067385e+18                0.000000   
50%                          -6.926075e+18            10010.000000   
75%                          -5.778284e+18            10013.000000   
max                          -4.611694e+18            10025.000000   

       gaTrackerData.profileConversionTrackingParams.trackingSRC  \
count                                       1.765700e+04           
mean                                        7.695416e+06           
std                                         1.449757e+06           
min                                         5.876256e+06          

In [4]:
pd.set_option('display.max_rows', None)  
pd.set_option('display.max_columns', None)

Como el dataset dispone de bastantes filas pandas solo me permite ver un numero limitado, asi que opte por desplegar todas las filas y poder verlas todas.

In [8]:
df.isnull().sum()

benefits.benefitRatingDecimal                                        0
benefits.comments                                                    1
benefits.highlights                                                  1
benefits.numRatings                                                  0
benefits.employerSummary                                        159541
breadCrumbs                                                          0
gaTrackerData.category                                               0
gaTrackerData.empId                                                  0
gaTrackerData.empName                                            18209
gaTrackerData.empSize                                            18209
gaTrackerData.expired                                                0
gaTrackerData.industry                                           47405
gaTrackerData.industryId                                             0
gaTrackerData.jobId.long                                           246
gaTrac

In [3]:
deleted_columns = ['breadCrumbs','gaTrackerData.category','gaTrackerData.empId','gaTrackerData.empSize','gaTrackerData.expired','gaTrackerData.industryId','gaTrackerData.jobId.long','gaTrackerData.jobId.int','gaTrackerData.locationId','gaTrackerData.locationType','gaTrackerData.pageRequestGuid.guid','gaTrackerData.pageRequestGuid.guidValid','gaTrackerData.pageRequestGuid.part1','gaTrackerData.pageRequestGuid.part2','gaTrackerData.sectorId','gaTrackerData.profileConversionTrackingParams.trackingCAT','gaTrackerData.profileConversionTrackingParams.trackingSRC','gaTrackerData.profileConversionTrackingParams.trackingXSP','gaTrackerData.jobViewTrackingResult.jobViewDisplayTimeMillis','gaTrackerData.jobViewTrackingResult.requiresTracking','gaTrackerData.jobViewTrackingResult.trackingUrl','header.adOrderId','header.advertiserType','header.applicationId','header.applyButtonDisabled','header.applyUrl','header.blur','header.coverPhoto','header.easyApply','header.employerId','header.expired','header.gocId','header.hideCEOInfo','header.locId','header.locationType','header.logo','header.logo2x','header.organic','header.overviewUrl','header.rating','header.saved','header.savedJobId','header.sgocId','header.sponsored','header.userAdmin','header.uxApplyType','header.featuredVideo','header.urgencyLabel','header.urgencyLabelForMessage','header.urgencyMessage','header.needsCommission','header.payHigh','header.payLow','header.payMed','header.payPeriod','header.salaryHigh','header.salaryLow','header.salarySource','job.eolHashCode','job.importConfigId','job.jobReqId.long','job.jobReqId.int','job.jobTitleId','job.listingId.long','job.listingId.int','map.address','map.postalCode','overview.allBenefitsLink','overview.allPhotosLink','overview.allReviewsLink','overview.allSalariesLink','overview.industryId','overview.sectorId','overview.stock','overview.mission','overview.allVideosLink','overview.competitors','overview.companyVideo','photos','rating.ceo.photo','rating.ceo.photo2x','rating.ceo.ratingsCount','rating.ceoApproval','rating.recommendToFriend','salary.country.cc3LetterISO','salary.country.ccISO','salary.country.continent.continentCode','salary.country.continent.continentName','salary.country.continent.id','salary.country.continent.new','salary.country.countryFIPS','salary.country.currency.currencyCode','salary.country.currency.defaultFractionDigits','salary.country.currency.displayName','salary.country.currency.id','salary.country.currency.negativeTemplate','salary.country.currency.new','salary.country.currency.positiveTemplate','salary.country.currency.symbol','salary.country.currencyCode','salary.country.defaultLocale','salary.country.defaultShortName','salary.country.employerSolutionsCountry','salary.country.id','salary.country.longName','salary.country.major','salary.country.name','salary.country.new','salary.country.population','salary.country.shortName','salary.country.tld','salary.country.type','salary.country.uniqueName','salary.country.usaCentricDisplayName','salary.currency.currencyCode','salary.currency.defaultFractionDigits','salary.currency.displayName','salary.currency.id','salary.currency.name','salary.currency.negativeTemplate','salary.currency.new','salary.currency.positiveTemplate','salary.currency.symbol','salary.lastSalaryDate','wwfu']

In [4]:
df_without_columns = df.drop(columns = deleted_columns)

Elimine estas columnas por los criterios que determine para mi objetivo principal, de las cuales vi que no eran necesarias.

In [14]:
df_without_columns.head()

benefits.benefitRatingDecimal  benefits.comments  benefits.highlights  \
0                            0.0             2801.0               2801.0   
1                            4.0             2802.0               2802.0   
2                            0.0             2803.0               2803.0   
3                            0.0             2804.0               2804.0   
4                            0.0             2805.0               2805.0   

   benefits.numRatings benefits.employerSummary  gaTrackerData.empName  \
0                    0                      NaN            Modus Group   
1                    3                      NaN   Boehringer Ingelheim   
2                    0                      NaN        Immobiliare.it    
3                    0                      NaN                  Codup   
4                    0                      NaN  Markelytics Solutions   

                  gaTrackerData.industry  \
0  Investment Banking & Asset Management   
1              Biotech & Pharmaceuticals   
2                                    NaN   
3                                    NaN   
4                             Consulting   

                              gaTrackerData.jobTitle gaTrackerData.location  \
0                 Biogas Project Development Manager                 Warsaw   
1  Quality Manager - Boehringer Ingelheim Healthc...                    NaN   
2     Senior Software Engineer (PHP, Elixir, Python)                 Milano   
3                                Senior SQA Engineer                Karachi   
4                                   Research Manager              Bengaluru   

        gaTrackerData.sector          header.employerName  \
0                    Finance                  Modus Group   
1  Biotech & Pharmaceuticals         Boehringer Ingelheim   
2                        NaN               Immobiliare.it   
3                        NaN                        Codup   
4          Business Services  MARKELYTICS SOLUTIONS INDIA   

                                     header.jobTitle header.location  \
0                 Biogas Project Development Manager          Warsaw   
1  Quality Manager - Boehringer Ingelheim Healthc...           Porto   
2     Senior Software Engineer (PHP, Elixir, Python)           Milan   
3                                Senior SQA Engineer         Karāchi   
4                                   Research Manager       Bengaluru   

  header.posted header.normalizedJobTitle  \
0  Oct 24, 2019                       NaN   
1   Nov 4, 2019                       NaN   
2  Oct 28, 2019                       NaN   
3   Nov 4, 2019                       NaN   
4   Nov 1, 2019                       NaN   

                                     job.description     job.discoverDate  \
0  We are the drivers of the future who offer a g...  2019-10-24T04:52:23   
1  <strong></strong>\n<br/><br/>\n<strong></stron...  2019-11-04T03:59:14   
2  Dottori.it &egrave; il motore di ricerca che t...  2019-10-28T12:48:55   
3  Codup is looking for a Senior SQA Engineer for...  2019-11-04T07:00:20   
4  <strong>Department:</strong> Research &amp; Op...  2019-11-01T13:40:48   

              job.jobSource map.country       map.employerName  map.lat  \
0            WorkableHR.com      Poland            Modus Group  52.2500   
1                     taleo         NaN   Boehringer Ingelheim  41.1500   
2  Indeed XML Feed Employer       Italy        Immobiliare.it   45.4667   
3    Indeed XML Feed Dradis          PK                  Codup  24.8667   
4  Indeed XML Feed Employer          IN  Markelytics Solutions  12.9670   

   map.lng map.location  overview.foundedYear                  overview.hq  \
0  21.0000       Warsaw                  1993           Vilnius, Lithuania   
1  -8.6167        Porto                  1885  Ingelheim am Rhein, Germany   
2   9.2000        Milan                     0                MILANO, Italy   
3  67.0500      Karāchi                     0            Karachi

In [5]:
column_formats = df_without_columns.dtypes
print(column_formats)

benefits.benefitRatingDecimal    float64
benefits.comments                float64
benefits.highlights              float64
benefits.numRatings                int64
benefits.employerSummary          object
gaTrackerData.empName             object
gaTrackerData.industry            object
gaTrackerData.jobTitle            object
gaTrackerData.location            object
gaTrackerData.sector              object
header.employerName               object
header.jobTitle                   object
header.location                   object
header.posted                     object
header.normalizedJobTitle         object
job.description                   object
job.discoverDate                  object
job.jobSource                     object
map.country                       object
map.employerName                  object
map.lat                          float64
map.lng                          float64
map.location                      object
overview.foundedYear               int64
overview.hq     

Tablas a utilizar: glassdoor_benefits_highlights.csv, glassdoor_salary_salaries.csv

In [6]:
df_without_columns.isnull().sum()

benefits.benefitRatingDecimal         0
benefits.comments                     1
benefits.highlights                   1
benefits.numRatings                   0
benefits.employerSummary         159541
gaTrackerData.empName             18209
gaTrackerData.industry            47405
gaTrackerData.jobTitle                0
gaTrackerData.location            70283
gaTrackerData.sector              47373
header.employerName                3245
header.jobTitle                       0
header.location                       5
header.posted                         0
header.normalizedJobTitle        136135
job.description                       4
job.discoverDate                      0
job.jobSource                         0
map.country                       50028
map.employerName                  18141
map.lat                               0
map.lng                               0
map.location                          5
overview.foundedYear                  0
overview.hq                       19367
